command/job_scale.go

```diff
@@ -0,0 +1,185 @@
+var _ cli.Command = &JobScaleCommand{}
+
+// JobScaleCommand implements cli.Command.
+type JobScaleCommand struct {
+	Meta
+}
+
+// Help satisfies the cli.Command Help function.
+func (j *JobScaleCommand) Help() string {
+	helpText := `
+Usage: nomad job scale [options] <args>
+
+  Perform a scaling action by altering the count within a job group.
+
+  Upon successful job submission, this command will immediately
+  enter an interactive monitor. This is useful to watch Nomad's
+  internals make scheduling decisions and place the submitted work
+  onto nodes. The monitor will end once job placement is done. It
+  is safe to exit the monitor early using ctrl+c.
+
+General Options:
+
+  ` + generalOptionsUsage() + `
+
+Scale Options:
+
+  -detach
+    Return immediately instead of entering monitor mode. After job scaling,
+    the evaluation ID will be printed to the screen, which can be used to
+    examine the evaluation using the eval-status command.
+
+  -verbose
+    Display full information.
+`
+	return strings.TrimSpace(helpText)
+}
+
+// Synopsis satisfies the cli.Command Synopsis function.
+func (j *JobScaleCommand) Synopsis() string {
+	return "Change the count of a Nomad job group"
+}
+
+func (j *JobScaleCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(j.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-detach":  complete.PredictNothing,
+			"-verbose": complete.PredictNothing,
+		})
+}
+
+// Name returns the name of this command.
+func (j *JobScaleCommand) Name() string { return "job scale" }
+
+// Run satisfies the cli.Command Run function.
+func (j *JobScaleCommand) Run(args []string) int {
+	var detach, verbose bool
+
+	flags := j.Meta.FlagSet(j.Name(), FlagSetClient)
+	flags.Usage = func() { j.Ui.Output(j.Help()) }
+	flags.BoolVar(&detach, "detach", false, "")
+	flags.BoolVar(&verbose, "verbose", false, "")
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	var jobString, countString, groupString string
+	args = flags.Args()
+
+	// It is possible to specify either 2 or 3 arguments. Check and assign the
+	// args so they can be validate later on.
+	if numArgs := len(args); numArgs < 2 || numArgs > 3 {
+		j.Ui.Error("Command requires at least two arguments and no more than three")
+		return 1
+	} else if numArgs == 3 {
+		groupString = args[1]
+		countString = args[2]
+	} else {
+		countString = args[1]
+	}
+	jobString = args[0]
+
+	// Convert the count string arg to an int as required by the API.
+	count, err := strconv.Atoi(countString)
+	if err != nil {
+		j.Ui.Error(fmt.Sprintf("Failed to convert count string to int: %s", err))
+		return 1
+	}
+
+	// Get the HTTP client.
+	client, err := j.Meta.Client()
+	if err != nil {
+		j.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+		return 1
+	}
+
+	// Detail the job so we can perform addition checks before submitting the
+	// scaling request.
+	job, _, err := client.Jobs().Info(jobString, nil)
```

Comment:
could this be done with `ScaleStatus` instead of `Info`? uses a smaller set of permissions that might be more appropriate for an automated pipeline?